consensus/mint-client/src/bin/main.rs

```diff
@@ -10,119 +10,158 @@ use mc_consensus_api::{
             let ch = ChannelBuilder::default_channel_builder(env).connect_to_uri(&node, &logger);
             let client_api = ConsensusClientApiClient::new(ch.clone());
             let blockchain_api = BlockchainApiClient::new(ch);
 
-            let signer = Ed25519Pair::from(signing_key);
-
-            let tombstone_block = tombstone.unwrap_or_else(|| {
-                let last_block_info = blockchain_api
-                    .get_last_block_info(&Empty::new())
-                    .expect("get last block info");
-                last_block_info.index + MAX_TOMBSTONE_BLOCKS - 1
-            });
-
-            let nonce = nonce.map(|n| n.to_vec()).unwrap_or_else(|| {
-                let mut rng = thread_rng();
-                let mut nonce: Vec<u8> = vec![0u8; NONCE_LENGTH];
-                rng.fill_bytes(&mut nonce);
-                nonce
-            });
-
-            let prefix = MintConfigTxPrefix {
-                token_id,
-                configs: vec![MintConfig {
-                    token_id,
-                    signer_set: SignerSet::new(vec![signer.public_key()], 1),
-                    mint_limit: 1000,
-                }],
-                nonce,
-                tombstone_block,
+            let tx = params
+                .try_into_mint_config_tx(|| {
+                    let last_block_info = blockchain_api
+                        .get_last_block_info(&Empty::new())
+                        .expect("get last block info");
+                    last_block_info.index + MAX_TOMBSTONE_BLOCKS - 1
+                })
+                .expect("failed creating tx");
+
+            let resp = client_api
+                .propose_mint_config_tx(&(&tx).into())
+                .expect("propose tx");
+            println!("response: {:?}", resp);
+        }
+
+        Commands::GenerateMintConfigTx { out, params } => {
+            let tx = params
+                .try_into_mint_config_tx(|| panic!("missing tombstone block"))
+                .expect("failed creating tx");
+
+            let json = to_string_pretty(&tx).expect("failed serializing tx");
+
+            fs::write(out, json).expect("failed writing output file");
+        }
+
+        Commands::SubmitMintConfigTx { node, tx_filenames } => {
+            // Load all txs.
+            let txs: Vec<MintConfigTx> = load_json_files(&tx_filenames);
+
+            // All tx prefixes should be the same.
+            if !txs.windows(2).all(|pair| pair[0].prefix == pair[1].prefix) {
+                panic!("All txs must have the same prefix");
+            }
+
+            // Collect all signatures.
+            let mut signatures = txs
+                .iter()
+                .flat_map(|tx| tx.signature.signatures())
+                .cloned()
+                .collect::<Vec<_>>();
+            signatures.sort();
+            signatures.dedup();
+
+            let merged_tx = MintConfigTx {
```

Comment:
Optional: Might this merge-txs logic be useful elsewhere? I imagine a `Mint[Config]Tx::combine(Vec)` kind of method would be helpful.

Reply:
I don't think this is necessary, I can't think of another place where this needs to happen. We can reconsider if that comes up in the future.